src/engine/strat_engine/thinpool/thinpool.rs

```diff
@@ -867,6 +945,56 @@ mod tests {
         assert!(pool.get_filesystem_by_uuid(fs_uuid).is_none());
     }
 
+    #[test]
+    pub fn loop_test_meta_expand() {
+        // This test requires more than 1 GiB.
+        loopbacked::test_with_spec(loopbacked::DeviceLimits::Range(2, 3), test_meta_expand);
+    }
+
+    #[test]
+    pub fn real_test_meta_expand() {
+        real::test_with_spec(real::DeviceLimits::Range(2, 3), test_meta_expand);
+    }
+
+    /// Verify that the meta device backing a ThinPool is expanded when meta
+    /// utilization exceeds the META_LOWATER mark, by creating a ThinPool with
+    /// a meta device smaller than the META_LOWATER.
+    fn test_meta_expand(paths: &[&Path]) -> () {
+        let pool_uuid = Uuid::new_v4();
+        let dm = DM::new().unwrap();
+        let mut mgr = BlockDevMgr::initialize(pool_uuid, paths, MIN_MDA_SECTORS, false).unwrap();
+        // Create a ThinPool with a very small meta device.
+        let mut thin_pool = ThinPool::new(pool_uuid,
+                                          &dm,
+                                          &ThinPoolSizeParams {
+                                               meta_size: MetaBlocks(16),
```

Comment:
```MetaBlocks(16)``` probably deservies a name in this method, since it gets used four times to mean the same thing.